cassandra/src/main/scala/io/zipkin/dependencies/spark/cassandra/ZipkinDependenciesJob.scala

```diff
@@ -111,28 +221,36 @@ case class ZipkinDependenciesJob(sparkMaster: String = ZipkinDependenciesJob.spa
       .filter { case (key, span) => span.isValid }
       .values
 
+    if (spans.isEmpty()) {
+      println(s"NO VALID SPANS FOUND IN THE TIME RANGE")
+    }
+
     val traces: RDD[Trace] = spans // guaranteed no duplicates of (trace id, span id)
       .map(span => (span.traceId, Trace(span.traceId, Map(span.id -> span)))) // (traceId -> Trace)
       .reduceByKey((t1, t2) => t1.mergeTrace(t2))
       .values
 
-    val aggregates: RDD[((String, String), Long)] = traces
+    case class Link(callCount: Long, instrumentation: Instrumentation) {
+      def +(link: Link): Link = {
+        Link(this.callCount + link.callCount, this.instrumentation + link.instrumentation)
+      }
+    }
+
+    val aggregates: RDD[((String, String), Link)] = traces
       .flatMap(_.getLinks)
-      .map { case (parent, child) => ((parent, child), 1L) } // start the count
+      .map { case (parent, child, instrumentation) => ((parent, child), Link(1L, instrumentation)) } // start the count
       .reduceByKey(_ + _) // add up the counts
 
     if (aggregates.isEmpty()) {
-      println(s"NO VALID SPANS FOUND IN THE TIME RANGE")
+      println(s"NO LINKS WERE GENERATED FROM SPANS")
     } else {
       val dependencies: DependenciesInfo = aggregates
-        .map { case ((parent: String, child: String), callCount: Long) =>
-        DependenciesInfo(Seq(DependencyLinkInfo(parent = parent, child = child, callCount = callCount)))
-      }
-        .reduce(_ + _) // merge DLs under one Dependencies object, which overrides +
+        .map { case ((parent: String, child: String), link: Link) =>
+        // TODO signal loss here - `instrumentation` is ignored
```

Comment:
ok so here's where you mean that there's no api impact, as the intermediate state is thrown out